src/main/java/ru/mercuriev/game/of/life/graph/GraphEngine.java

```diff
@@ -0,0 +1,33 @@
+package ru.mercuriev.game.of.life.graph;
+
+import ru.mercuriev.game.of.life.graph.cell.Cell;
+
+import java.util.Arrays;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ * Engine interface implementation
+ * Uses Cell uder the hood to calculate next generation
+ *
+ * @author paul
+ */
+// TODO implement Engine interface
+public class GraphEngine {
+
+    // TODO @Override
+    public int[][] next(int[][] currentGeneration) {
+
+        // int[][] -> Stream<IntStream>
+        Stream<IntStream> input = Arrays.stream(currentGeneration)
```

Comment:
Yes, it was the first implementation. But using reference type instead of primitive will cause the change of the ```next()``` method signature, and we've decided to use common inetraface for all engines. Another option us to cast ```int[][]``` to ```Integer[][]``` *only* to use ```Stream<Strea<Integer>>```. Moreover, using stream of promitives is not an antipattern.